### ISSUE TITLE
OPNsense/OpenVPN using SSL + user authentication documentation required

### ISSUE THREAD

**Issue by noseshimself** [opnsense/docs#252]
**Is your feature request related to a problem? Please describe.**
As authentication scripts are not standardized it is not clear which user name and password are used for additional authentication in this case. It is obviously not the account the certificate is connected to.

**Describe the solution you'd like**
1) An exact description in the OPNsense manual section "Setup SSL VPN Road Warrior" (maybe renaming it might be a good idea because the term "Road Warrior" is not showing up anywhere in the menus) explaining the relationship between the account's user name, the Descriptive Name and the Common Name entries in the linked certificate and which of them is the User Name OpenVPN will demand for authentication.

2) Add some reminders to the info/help texts in the OpenVPN section.

**Describe alternatives you've considered**
Finding out by reading the source code for hours.

**Comment by AdSchellevis** [comment]
Not sure what you mean, authentication is quite standardised in OPNsense (https://docs.opnsense.org/development/components/authentication.html).

 I've explained the relation between the certificate and the user a bit more in the existing document (https://github.com/opnsense/docs/commit/ad587d66c33bc34f5300b9ae35e18c79dc69e800), but in general the document still seems to be quite accurate. 

The term road warrior likely won't come back in the product itself, since it's more of a usage scenario than an actual option, the intro explains what it is already https://docs.opnsense.org/manual/vpnet.html. 

A more in-depth list of options for OpenVPN might be practical, but I don't expect that to be very high on anyones list at the moment.

If you have suggestions we can review, just open a PR and we'll take it from there. There's always room for improvements.

**Comment by seamusdemora** [comment]
@AdSchellevis : I get the OP's confusion. I also feel the term "Road Warrior" is a bit misleading or confusing... perhaps if you'd ever watched Mel Gibson's *classic movie* by that name, you'd understand  why I say this :) Perhaps "External User" would be better? Idioms like "Road Warrior" can be misleading for some. 

Also, while I think the document he referenced is a good one (I used it a few years ago to set up my VPN), I do think it would benefit from a **brief** explanation of how all of the certificates, etc fit into the overall authentication process, and why they're necessary. 

Finally, I'd like to say these comments are offered without criticism of the documentation. Overall, I believe that OPNsense is an incredible project, and I am grateful to you all.

**Comment by noseshimself** [comment]
> Not sure what you mean, authentication is quite standardised in OPNsense (https://docs.opnsense.org/development/components/authentication.html).

I would have expected that the "user name" the client has to provide would have to be the user name from the user database and not something provided inside the certificate. That's a bit counter-intuitive and now I'm wondering where the password is coming from; it's obviously not the login's password or something is wrong withseveral implementations of OpenVPN I have been testing it with.

> I've explained the relation between the certificate and the user a bit more in the existing document ([ad587d6](https://github.com/opnsense/docs/commit/ad587d66c33bc34f5300b9ae35e18c79dc69e800)), but in general the document still seems to be quite accurate.

I'm going to take a look at it.

> The term road warrior likely won't come back in the product itself, since it's more of a usage scenario than an actual option, the intro explains what it is already https://docs.opnsense.org/manual/vpnet.html.

Let me rephrase that: Why isn't it called "OpenVPN client/server setup"; the term "Road Warrior" is not connected to using OpenVPN. Most people I know who would qualify for this term seem to be using Zerotier for some unknown reason.

> If you have suggestions we can review, just open a PR and we'll take it from there. There's always room for improvements.

I have spent hours not getting Viscosity to work and because all I can find in the log is an authentication failure I'm not understanding because the mechanism itself is not documented. And all examples I could find just left that step out.

**Comment by noseshimself** [comment]
> @AdSchellevis : I get the OP's confusion. I also feel the term "Road Warrior" is a bit misleading or confusing... perhaps if you'd ever watched Mel Gibson's _classic movie_ by that name, you'd understand why I say this :) Perhaps "External User" would be better? Idioms like "Road Warrior" can be misleading for some.

What I'm missing is the connection between the protocol/application and the usage scenario. It could be done using IPsec or any of the other available VPN tools as well.

**Comment by AdSchellevis** [comment]
You probably have to contribute some (draft) text yourself in a PR, as I said, the current text seem to help for quite some people, I've reviewed the document a couple of days ago and really don't see a lot of improvements in it. Documenting all options might help, although in my experience these lists tend to confuse people even more. Our documentation currently only covers some usage scenarios, including remote users, which in my opinion is helpful for most people, yet not perfect (I guess nothing is).

Long story short, I'm offering to review your contribution and help this further.

**Comment by seamusdemora** [comment]
@AdSchellevis :  With respect, that sounds a bit like the line from an old Western movie: "We'll give you a fair trial before we hang you." LOL Sorry - but you started it with this "Road Warrior" business :) 

But to get to the serious part of my comment: It seems there's very little in the way of DNS configuration documented in the VPN/RoadWarrior material. This has confounded me more than any other aspect of VPN setup.

**Comment by AdSchellevis** [comment]
@seamusdemora it isn't and I didn't start with the "Road Warrior" business either. Just noting that people are free to contribute to our open source project, it's just not enough to ask for "better", since different people will interpret better in different ways. I myself am not planning to work on this topic at the moment (other than the changes in https://github.com/opnsense/docs/commit/ad587d66c33bc34f5300b9ae35e18c79dc69e800), which doesn't mean someone else can't do so, in which case we will review the proposed changes with an open mind.

**Comment by mimugmail** [comment]
Road Warrior came from FreeSWAN which was THE IPsec implementation and was already mentioned in early 2000:

https://www.freeswan.org/freeswan_trees/freeswan-2.00/CHANGES

**Comment by seamusdemora** [comment]
Well that's fine... that document is ca. 2003, and the movie came out in 1981 or '82 I think. I'm not talking about historical precedents - just saying that "Road Warrior" label sounds a little worn. But hey - you wrote it, you can call it whatever you like. And sorry if I touched a sensitive nerve. Please forget I mentioned it.
